guides/keras_hub/object_detection_retinanet.py

```diff
@@ -0,0 +1,796 @@
+eval_ds = preprocess_tfds(eval_ds)
+
+"""
+### Let's visualize batch of training data
+"""
+record = next(iter(train_ds.shuffle(100).take(1)))
+keras.visualization.plot_bounding_box_gallery(
+    record["images"],
+    bounding_box_format=gt_bbox_format,
+    y_true=record["bounding_boxes"],
+    scale=3,
+    rows=2,
+    cols=2,
+    class_mapping=INDEX_TO_CLASS,
+)
+
+"""
+### Decoded TFDS record to a tuple for keras-hub
+"""
+train_ds = train_ds.map(convert_to_tuple, num_parallel_calls=tf.data.AUTOTUNE)
+train_ds = train_ds.prefetch(tf.data.AUTOTUNE)
+
+eval_ds = eval_ds.map(convert_to_tuple, num_parallel_calls=tf.data.AUTOTUNE)
+eval_ds = eval_ds.prefetch(tf.data.AUTOTUNE)
+
+"""
+## Configure RetinaNet Model
+Configure the model with `backbone`, `num_classes` and `preprocessor`.
+Use callbacks for recording logs and saving checkpoints.
+"""
+
+
+def get_callbacks(experiment_path: str):
+    """Creates a list of callbacks for model training.
+
+    Args:
+      experiment_path (str): Path to the experiment directory.
+
+    Returns:
+      List of keras callback instances.
+    """
+    # models_path = os.path.join(experiment_path, "tf_models")
+    tb_logs_path = os.path.join(experiment_path, "logs")
+    ckpt_path = os.path.join(experiment_path, "weights")
+    return [
+        keras.callbacks.BackupAndRestore(ckpt_path, delete_checkpoint=False),
+        keras.callbacks.TensorBoard(
+            tb_logs_path,
+            update_freq=1,
+        ),
+        keras.callbacks.ModelCheckpoint(
+            ckpt_path + "/{epoch:04d}-{val_loss:.2f}.weights.h5",
+            save_best_only=True,
+            save_weights_only=True,
+            verbose=1,
+        ),
+    ]
+
+
+"""
+## Load backbone weights and preprocessor config
+
+Let's use the "retinanet_resnet50_fpn_coco" pretrained weights as the backbone
+model, applying its predefined configuration from the preprocessor of the
+"retinanet_resnet50_fpn_coco" preset.
+Define a RetinaNet object detector model with the backbone and preprocessor
+specified above, and set `num_classes` to 20 to represent the object categories
+from Pascal VOC.
+Finally, compile the model using Mean Absolute Error (MAE) as the box loss.
+"""
+
+backbone = keras_hub.models.Backbone.from_preset("retinanet_resnet50_fpn_coco")
+
+preprocessor = keras_hub.models.RetinaNetObjectDetectorPreprocessor.from_preset(
+    "retinanet_resnet50_fpn_coco"
+)
+model = keras_hub.models.RetinaNetObjectDetector(
+    backbone=backbone, num_classes=len(CLASSES), preprocessor=preprocessor
+)
+model.compile(box_loss=keras.losses.MeanAbsoluteError(reduction="sum"))
+
+"""
+## Train the model
+Now that the object detector model is compiled, let's train it using the
+training and validation data we created earlier.
+For demonstration purposes, we have used a small number of epochs. You can
+increase the number of epochs to achieve better results.
+
+**Note:** The model trained on L4 GPU, while training on T4 it takes
```

Comment:
this note doesn't add much. what is "significant time"?

Maybe say--training will take x hours on a free tier colab gpu (t4)